graal-languages-core/src/main/java/io/micronaut/graallanguages/ContextProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.graallanguages;
+
+import io.micronaut.core.annotation.Experimental;
+import org.graalvm.polyglot.Context;
+
+/**
+ * Wrapper around {@link Context} since it is a final class and cannot be proxied after its instantiation in a factory.
+ * @author Sergio del Amo
+ * @since 1.1.0
+ */
+@Experimental
+public class ContextProvider {
```

Comment:
This looks nice. There is still one issue: you can get Java objects that wrap JavaScript/Python/... objects that are associated with given context. Example:
```
Value fun = useContext(ctx -> ctx.eval("js", "x => x + 1"));
fun.execute(42); // <-- this will execute the function in the `Context` where it originated from
```

thinking about this, I am not sure there can ever be some "compile-time safety net" for this, users will have to understand the constraints and make sure they do not hold onto `Value` instances from the `Context` outside of the lambda.

@mikehearn used try-with-resources for his `Context` pool (but that doesn't solve this issue either).

Reply:
Techically we can check what type is being returned from the method, but that would have to be an implementation specific check

Reply:
Side note: it is even trickier with the proxy feature:
```
Value pythonClassInstance = ctx.eval("python", """
   class MyPythonClass:
      def foo(x): return x+1

   MyPythonClass() # create instance
""")

MyJavaInterface javaProxy = pythonClassInstance.as(MyJavaInterface.class);
// now javaProxy looks like plain Java object, but internally it is backed by `Value`

assert javaProxy.foo(42) == 43
```